worker/uniter/uniter.go

```diff
@@ -941,20 +941,25 @@ func (u *Uniter) reportHookError(hookInfo hook.Info) error {
 	// hook is interrupted (e.g. unit agent crashes), rather than immediately
 	// after attempting a runHookOp.
 	hookName := string(hookInfo.Kind)
+	hookMessage := string(hookInfo.Kind)
 	statusData := map[string]interface{}{}
 	if hookInfo.Kind.IsRelation() {
 		statusData["relation-id"] = hookInfo.RelationId
 		if hookInfo.RemoteUnit != "" {
 			statusData["remote-unit"] = hookInfo.RemoteUnit
 		}
 		relationName, err := u.relationStateTracker.Name(hookInfo.RelationId)
-		if err != nil {
+		if errors.IsNotFound(err) {
+			hookMessage = fmt.Sprintf("%s: %v", hookInfo.Kind, err)
+		} else if err != nil {
 			return errors.Trace(err)
+		} else {
+			hookName = fmt.Sprintf("%s-%s", relationName, hookInfo.Kind)
+			hookMessage = hookName
```

Comment:
We could even consider making a hook error message for any error and reporting that via setAgentStatus() below. We are inside a method where something by definition has already gone wrong. Should we bounce the agent if there's an error dealing with an error?